static/app/views/settings/organizationIntegrations/sentryAppExternalForm.tsx

```diff
@@ -419,6 +422,7 @@ export class SentryAppExternalForm extends Component<Props, State> {
         }}
         onSubmitError={this.onSubmitError}
         onFieldChange={this.handleFieldChange}
+        preventFormResetOnUnmount
```

Comment:
Because we're managing form state out of band from the form, we don't want to the unmounting of the form to reset the state. This component will handle it itself.

We had to change this for this to work because the unmount call-order is different between useEffect and class components, so the reset was happening AFTER this component was resetting the initial state up in the `resetStateFromProps`